src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -355,13 +356,12 @@ private void BuildFallbackBlock(TagHelperContent builder)
                 // indicating that the primary stylesheet failed to load.
                 builder
                     .AppendHtml("<script>")
-                    .AppendHtml(
-                        string.Format(
-                            CultureInfo.InvariantCulture,
-                            JavaScriptResources.GetEmbeddedJavaScript(FallbackJavaScriptResourceName),
-                            JavaScriptEncoder.Encode(FallbackTestProperty),
-                            JavaScriptEncoder.Encode(FallbackTestValue),
-                            JavaScriptStringArrayEncoder.Encode(JavaScriptEncoder, fallbackHrefs)))
+                    .AppendFormat(
+                        CultureInfo.InvariantCulture,
+                        JavaScriptResources.GetEmbeddedJavaScript(FallbackJavaScriptResourceName),
+                        new HtmlString(JavaScriptEncoder.Encode(FallbackTestProperty)),
+                        new HtmlString(JavaScriptEncoder.Encode(FallbackTestValue)),
+                        new HtmlString(JavaScriptStringArrayEncoder.Encode(JavaScriptEncoder, fallbackHrefs)))
```

Comment:
If you want to follow up on this issue one way or another in a separate PR that's fine by me. Either way, a .Format-based API will never be as good as just calling .Append a bunch of times